libraries/ms-common/src/x86/i2c.c

```diff
@@ -1,23 +1,62 @@
 #include "i2c.h"
+#include "log.h"
 
 StatusCode i2c_init(I2CPort i2c, const I2CSettings *settings) {
-  return STATUS_CODE_UNIMPLEMENTED;
+  if (i2c >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
+  } else if (settings->speed >= NUM_I2C_SPEEDS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C speed.");
+  }
+  LOG_DEBUG("Note this is an x86 version of I2C\n");
+  return STATUS_CODE_OK;
 }
 
 StatusCode i2c_read(I2CPort i2c, I2CAddress addr, uint8_t *rx_data, size_t rx_len) {
-  return STATUS_CODE_UNIMPLEMENTED;
+  if (i2c >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
+  }
+  LOG_DEBUG("Reading %ld bytes from I2C\n", rx_len);
+  for (size_t i = 0; i < rx_len; i++) {
+    // Insert dummy data
+    rx_data[i] = i % 2;
+  }
+  return STATUS_CODE_OK;
 }
 
 StatusCode i2c_write(I2CPort i2c, I2CAddress addr, uint8_t *tx_data, size_t tx_len) {
-  return STATUS_CODE_UNIMPLEMENTED;
+  if (i2c >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
+  }
+  LOG_DEBUG("Sending %ld bytes over I2C\n", tx_len);
+  for (size_t i = 0; i < tx_len; i++) {
+    LOG_DEBUG("0x%x", tx_data[i]);
+  }
+  LOG_DEBUG("\n");
```

Comment:
`LOG_DEBUG` inserts a prefix like `[0] libraries/src/ms-common/i2c.c:50:` before its argument every time, so this won't work for putting the bytes on one line.

To put everything on one line, I'd recommend removing the \n from the "Sending %ld bytes over I2C" log and then using printf for the other logs.